app/models/customer.rb

```diff
@@ -1,3 +1,9 @@
 class Customer < ApplicationRecord
+  has_many :invoices
 
-end
\ No newline at end of file
+  def self.top_customers
+    customers = joins(invoices: :transactions).where(transactions: {result: 'success'})
+    hash = customers.group("(first_name || ' ' || last_name)").count
```

Comment:
is there a reason this line isn't just chained onto the one above it? not really an issue just curious